src/UnitTests/XpressionMapper.ForPath.Tests.cs

```diff
@@ -0,0 +1,236 @@
+                     CustomerAge = 31
+                }
+            }.AsQueryable<OrderDto>();
+        }
+
+        private static IQueryable<OrderDto> Orders { get; set; }
+        private static IQueryable<DerivedDataModel> DataObjects { get; set; }
+
+        private void SetupAutoMapper()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.AddProfiles(typeof(ForPathCustomerProfile));
+            });
+
+            mapper = config.CreateMapper();
+        }
+
+        static IMapper mapper;
+    }
+
+    public class RootModel
+    {
+        public int ID { get; set; }
+        public string Title { get; set; }
+        public NestedModel Nested { get; set; }
+    }
+
+    public class NestedModel
+    {
+        public int NestedID { get; set; }
+        public string NestedTitle { get; set; }
+        public string NestedTitle2 { get; set; }
+    }
+
+    public class DerivedModel : RootModel
+    {
+        public string DescendantField { get; set; }
+    }
+
+    public class DataModel
+    {
+        public int ID { get; set; }
+        public string Title { get; set; }
+
+        public int OtherID { get; set; }
+        public string Title2 { get; set; }
+    }
+
+    public class DerivedDataModel : DataModel
+    {
+        public string DescendantField { get; set; }
+    }
+
+    public class Order
+    {
+        public CustomerHolder CustomerHolder { get; set; }
+        public int Value { get; }
+    }
+
+    public class CustomerHolder
+    {
+        public Customer Customer { get; set; }
+    }
+
+    public class Customer
+    {
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public decimal? Total { get; set; }
+        public decimal Age { get; set; }
+    }
+
+    public class CustomerDto
+    {
+        public string Name { get; set; }
+        public decimal? Total { get; set; }
+    }
+
+    public class OrderDto
+    {
+        public string CustomerAddress { get; set; }
+        public decimal CustomerAge { get; set; }
+        public CustomerDto Customer { get; set; }
+    }
+
+    public class ForPathCustomerProfile : Profile
+    {
+        public ForPathCustomerProfile()
+        {
+            CreateMap<DerivedDataModel, DerivedModel>()
+                .ForPath(d => d.Nested.NestedTitle, opt => opt.MapFrom(src => src.Title))
+                .ForPath(d => d.Nested.NestedTitle2, opt => opt.MapFrom(src => src.Title2));
+
+            CreateMap<OrderDto, Order>()
+                .ForPath(o => o.CustomerHolder.Customer.Name, o => o.MapFrom(s => s.Customer.Name))
+                .ForPath(o => o.CustomerHolder.Customer.Total, o => o.MapFrom(s => s.Customer.Total))
+                .ForPath(o => o.CustomerHolder.Customer.Address, o => o.MapFrom(s => s.CustomerAddress))
+                .ForPath(o => o.CustomerHolder.Customer.Age, o => o.MapFrom(s => s.CustomerAge));
+
+            CreateMissingTypeMaps = true;
```

Comment:
Please no :)

Reply:
Made it nullable :)

Reply:
OK, I should have said what I meant :) I have an issue with setting CreateMissingTypeMaps.

Reply:
Ah -  got it.